tests/lib/collectors/gcloud_e2e.py

```diff
@@ -161,51 +150,42 @@ def test_end_to_end_other_disk(self):
-
-    # Attach the disk copy to the analysis VM
-    gcloud_collector.Process()
-    self.__assertInstanceAndDisk()
-
-  def __assertCollectorIsInitialized(self, gcloud_collector):
-    self.assertEqual(self.test_state.errors, [])
-    self.assertEqual(gcloud_collector.analysis_project.project_id,
-                     self.project_id)
-    self.assertEqual(gcloud_collector.remote_project.project_id,
-                     self.project_id)
-    self.assertEqual(gcloud_collector.remote_instance_name,
-                     self.instance_to_analyse)
-    self.assertEqual(gcloud_collector.all_disks, False)
-
-  def __assertInstanceAndDisk(self):
-    self.assertEqual(
-        self.test_state.output[0][0], 'gcp-forensics-vm-fake-incident-id')
-    self.assertIsInstance(self.test_state.output[0][1], gcp.GoogleComputeDisk)
-    self.assertTrue(self.test_state.output[0][1].name.startswith(
-        'incidentfake-incident-id'))
-    self.assertTrue(self.test_state.output[0][1].name.endswith('-copy'))
-    self.assertIsInstance(self.analysis_vm, gcp.GoogleComputeInstance)
-    self.assertEqual(self.analysis_vm.name, 'gcp-forensics-vm-fake-incident-id')
-    self.assertEqual(
-        self.analysis_vm.list_disks(),
-        ['gcp-forensics-vm-fake-incident-id',
-         self.test_state.output[0][1].name])
+    # Attach the disk_to_forensic copy to the analysis VM
+    self.gcloud_collector.Process()
 
-  def tearDown(self):
-    project = gcp.GoogleCloudProject(project_id=self.project_id,
-                                     default_zone=self.zone)
+    # The forensic instance should be live in the analysis GCP project and
+    # the disk should be attached
+    analysis_vm_name = self.test_state.output[0][0]
+    attached_disk_name = self.test_state.output[0][1].name
+    try:
+      operation = self.gcp.gce_api().instances().get(
+          project=self.project_id,
+          zone=self.zone,
+          instance=analysis_vm_name).execute()
+      result = self.gcp.gce_operation(operation, zone=self.zone)
+
+      self.assertEqual(result['name'], analysis_vm_name)
+      for disk in result['disks']:
+        if disk['source'].split("/")[-1] == attached_disk_name:
+          return
+      self.fail('Error: could not find the disk {0:s} in instance {1:s}'.format(
+          attached_disk_name, analysis_vm_name
+      ))
+    except RuntimeError as exception:
+      self.fail('Error: could not find the forensic instance in the analysis '
+                'project: {0:s}'.format(str(exception)))
 
-    disks = self.analysis_vm.list_disks()
+  def tearDown(self):
+    project = self.gcloud_collector.analysis_project
+    disks = self.gcloud_collector.analysis_vm.list_disks()
 
     # delete the created forensics VMs
     log.info('Deleting analysis instance: {0:s}.'.format(
-        self.analysis_vm.name))
+        self.gcloud_collector.analysis_vm.name))
     operation = project.gce_api().instances().delete(
```

Comment:
For consistency it's better that you use the `self.gcp` attribute you defined in L70